subsys/usb/class/cdc_acm.c

```diff
@@ -745,6 +755,16 @@ static void cdc_acm_irq_callback_set(struct device *dev,
 	dev_data->cb_data = cb_data;
 }
 
+#if defined(CONFIG_CDC_ACM_ARDUINO_RESET_SUPPORT)
+void cdc_acm_arduino_reset_callback_set(
+	struct device *dev, cdc_acm_arduino_reset_callback_t callback)
+{
+	struct cdc_acm_dev_data_t *const dev_data = DEV_DATA(dev);
```

Comment:
Verify that `dev->driver_api` is equal to `cdc_acm_driver_api` and return an error if it isn't.  Otherwise passing the wrong device will corrupt something else's state.

Reply:
Done.